.github/workflows/continous-integration-docker.yml

```
@@ -385,7 +384,7 @@ jobs:
     # to also run on Mergify head branches,
     # add `|| (github.event_name == 'push' && startsWith(github.head_ref, 'mergify/merge-queue/'))`:
     # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-based-on-the-head-or-base-branch-of-a-pull-request-1
-    if: ${{ (!cancelled() && !failure() && github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true') || !fromJSON(needs.get-available-disks.outputs.lwd_tip_disk) }}
+    if: ${{ (!cancelled() && !failure() && (github.event_name == 'push' && github.ref_name == 'main') && github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true') || !fromJSON(needs.get-available-disks.outputs.lwd_tip_disk) }}
```

I'm wondering about the logic here:
- `!cancelled() && !failure()` - we must have success or skip for previous jobs, regardless of any other conditions, so I think these conditions need to be `&&` with all the rest of the conditions or triggers
- `github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true'` - these inputs will never be true on a push event, they are only set for manual workflow runs, so we can remove them
- I think we'll want to be able to manually trigger this job, like the zebrad checkpoint and full sync jobs (I'll open a PR for this)

So we might want to use the same conditions as the zebrad full sync, with an extra `!cancelled() && !failure()` at the start:
```suggestion
    if: ${{ !cancelled() && !failure() && ((github.event_name == 'push' && github.ref_name == 'main') || !fromJSON(needs.get-available-disks.outputs.lwd_tip_disk)) }}
```

Here are the zebrad full sync conditions, for comparison:
https://github.com/ZcashFoundation/zebra/blob/8646a03e03b9dda190c46f05fd64631a99c9da9d/.github/workflows/continous-integration-docker.yml#L321